app/controllers/works_controller.rb

```diff
@@ -0,0 +1,117 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update, :destroy, :upvote]
+  before_action :find_session, only: [:upvote] #TODO: I don't see why I would put this in app. controller if the heroku model only requires login for upvoting? So I put it here
```

Comment:
The reason you want it in the `ApplicationController` is because you want to display the username at the top of every page if it exists.